src/MOCKernel.cpp

```diff
@@ -0,0 +1,159 @@
+#include "MOCKernel.h"
+#include "TrackGenerator.h"
+
+/**
+ * @brief Constructor for the MOCKernel assigns default values
+ * @details The count for the number of segments traversed is set to zero and
+ *          the maximum optical path length is retrieved from the
+ *          TrackGenerator
+ * @param track_generator the TrackGenerator used to pull relevant tracking
+ *        data from
+ * @param row_num the row index into the temporary segments matrix
+ */
+MOCKernel::MOCKernel(TrackGenerator* track_generator, int row_num) {
+  _count = 0;
+  _max_tau = track_generator->retrieveMaxOpticalLength();
+}
+
+
+/**
+ * @brief Constructor for the VolumeKernel assigns default values, calls
+ *        the MOCKernel constructor, and pulls refernces to FSR locks and FSR
+ *        volumes from the provided TrackGenerator.
+ * @param track_generator the TrackGenerator used to pull relevant tracking
+ *        data from
+ * @param row_num the row index into the temporary segments matrix
```

Comment:
Is there really a segments "matrix"? I find `row_num` to be a non-intuitive name for a variable here, but perhaps that is just because I'm not as familiar with your OTF stuff to come.